library/src/main/cpp/decoder_jxl.cpp

```diff
@@ -0,0 +1,201 @@
+      }
+
+      pixels->resize(buffer_size);
+      void *pixels_buffer = (void *)pixels->data();
+      if (JXL_DEC_SUCCESS != JxlDecoderSetImageOutBuffer(dec.get(), &format,
+                                                         pixels_buffer,
+                                                         buffer_size)) {
+        throw std::runtime_error("JxlDecoderSetImageOutBuffer failed");
+      }
+    } else if (status == JXL_DEC_FULL_IMAGE) {
+    } else if (status == JXL_DEC_SUCCESS) {
+      break;
+    } else {
+      LOGW("Unexpected decoder status");
+      break;
+    }
+  }
+}
+
+ImageInfo JpegxlDecoder::parseInfo() {
+  Rect bounds;
+  JxlBasicInfo jxl_info;
+
+  if (cropBorders) {
+    std::vector<uint8_t> pixels;
+    decodeRGB(stream->bytes, stream->size, 0, &pixels, &jxl_info);
+
+    uint32_t num_pixels = jxl_info.xsize * jxl_info.ysize;
+    uint8_t *pixels_buffer = (uint8_t *)pixels.data();
+    uint32_t num_channels =
+        (jxl_info.alpha_bits > 0 ? 1 : 0) + jxl_info.num_color_channels;
+
+    if (num_channels == 2) {
+      for (int x = 0; x < num_pixels * num_channels; x += num_channels) {
+        pixels_buffer[x / num_channels] = pixels_buffer[x];
+      }
+    } else if (num_channels == 3 || num_channels == 4) {
+      for (int x = 0; x < num_pixels * num_channels; x += num_channels) {
+        uint8_t r = pixels_buffer[x + 0];
+        uint8_t g = pixels_buffer[x + 1];
+        uint8_t b = pixels_buffer[x + 2];
+        int gray = (r * 0.2126 + g * 0.7152 + b * 0.0722) + 0.5;
+        if (gray > 255) {
+          gray = 255;
+        }
+        pixels_buffer[x / num_channels] = (uint8_t)gray;
+      }
+    }
+
+    bounds = findBorders(pixels.data(), jxl_info.xsize, jxl_info.ysize);
+  } else {
+    auto dec = JxlDecoderMake(NULL);
+
+    if (JXL_DEC_SUCCESS !=
+        JxlDecoderSubscribeEvents(dec.get(), JXL_DEC_BASIC_INFO)) {
+      throw std::runtime_error("JxlDecoderSubscribeEvents failed");
+    }
+
+    JxlDecoderSetInput(dec.get(), stream->bytes, stream->size);
+
+    for (;;) {
+      JxlDecoderStatus status = JxlDecoderProcessInput(dec.get());
+
+      if (status == JXL_DEC_ERROR) {
+        throw std::runtime_error("Decoder error");
+      } else if (status == JXL_DEC_NEED_MORE_INPUT) {
+        throw std::runtime_error("Already provided all input");
+      } else if (status == JXL_DEC_BASIC_INFO) {
+        if (JXL_DEC_SUCCESS != JxlDecoderGetBasicInfo(dec.get(), &jxl_info)) {
+          throw std::runtime_error("JxlDecoderGetBasicInfo failed");
+        }
+        break;
+      } else if (status == JXL_DEC_SUCCESS) {
+        break;
+      } else {
+        LOGW("Unexpected decoder status");
+        break;
+      }
+    }
+
+    bounds = {
+        .x = 0, .y = 0, .width = jxl_info.xsize, .height = jxl_info.ysize};
+  }
+
+  return ImageInfo{.imageWidth = jxl_info.xsize,
+                   .imageHeight = jxl_info.ysize,
+                   .isAnimated = (bool)jxl_info.have_animation,
```

Comment:
This should return false, at least for now, because tachiyomi will try to use Coil to show the animated image, but it won't recognize this format.

We should then try to decode the first frame (I don't know if it already does that).